cli/tests/testdata/test/steps/invalid_usage.out

```diff
@@ -9,9 +9,6 @@ top level missing await ... FAILED ([WILDCARD])
 inner missing await ...
   step ...
     inner ... pending ([WILDCARD])
-      error: Error: Parent scope completed before test step finished execution. Ensure all steps are awaited (ex. `await t.step(...)`).
-          at [WILDCARD]
-          at async TestContext.step [WILDCARD]
```

Comment:
Why was this removed?

Reply:
This is a redundant error, the parent's error combined with `... pending` represents this. It appears inconsistently as well. For example, turning sanitizers on for `inner missing await` strangely gets rid of it. And it's only reported if the root test scope happens to be open still. Combine that with it creating the need for the error value contained in `TestStepResult::Pending()`, no reason to have it.

Reply:
`pending` isn't very instructive for users though. Can we output an instructive message still?

> It appears inconsistently as well.

We should fix that bug in that case.

Reply:
> Can we output an instructive message still?

Yes, when a `pending` test step result is printed in rust we can follow that up with any message along with that step's location.

I also wanted to follow up with a change that moved all validation errors to structured error variants which are formatted in Rust instead. We can print a nicely formatted location next to them rather than those loud JS error stacks. It's probably better to do it all at once in this PR.

Reply:
Done 
![image](https://user-images.githubusercontent.com/29990554/226141227-7565bce1-1759-4a0c-b116-0a6a040bf1e2.png)